### ISSUE TITLE
Clearing all resources from FhirBase

### ISSUE THREAD

**Issue by mschielmann** [fhirbase/fhirbase#119]
Would there be a possibility to provide a script/function to clear all resources from fhirbase so that after running it, the base would be in the same state as just after initiation?

I've created an example script:
```
DO $$
DECLARE
	v_delete_command TEXT;
	v_excluded_tables TEXT[];
BEGIN
	v_excluded_tables = ARRAY['ConceptMap', 'OperationDefinition', 'Resource', 
				'SearchParameter', 'StructureDefinition', 'ValueSet'];

	FOR v_delete_command IN
		SELECT
			'DELETE FROM public.' || lower(logical_id) || '; ' ||
			'DELETE FROM public.' || lower(logical_id) || '_history; '
		FROM
			public.structuredefinition
		WHERE
			kind='resource'
			AND NOT v_excluded_tables @> ARRAY[logical_id]
	LOOP
		EXECUTE v_delete_command;
	END LOOP;
END;
$$
```
The problem is that the tables like ContactMap, SearchParameters and so on inherit from Resource and are in the same schema. So the concern is that it'll stop working if there is a change in one of the six tables listed in `v_excluded_tables` or if a new table should be added to the excluded list.

Would it be possible to incorporate this kind of functionality into the fhirbase?

**Comment by niquola** [comment]
What is the use case for this feature? 
Would not it be simpler just to drop and regenerate database?

**Comment by mschielmann** [comment]
We're clearing fhirbase on our test suites as deleting resources leaves trace in _history tables. For some test cases we need to be sure that the db is clear.
Cleaning db with DELETE query is faster and simpler than rebuilding DB/restarting docker.
This is an enhancement for the developers use mainly so I'll understand if you won't find it feasible for the  fhirbase core.

**Comment by niquola** [comment]
Usualy such tests are done in transaction and after test you can rollback transaction and get clear db.
The problem is only with integration and acceptance tests.
I think we can add function fhir.truncate('<ResourceName>') to clear specific resource tables.

**Comment by mschielmann** [comment]
As you mentioned - the problem for us is with the integration and acceptance tests. I see this as a nice-to-have enhancement. Yet I'm looking for something like global-clear rather than:
```
delete from public.patient;
delete from public.patient_history;
```
or
```
fhir.truncate('patient')
```
Unless fair.truncate(null) could clear all but ConceptMap, ValueSet etc.?

**Comment by niquola** [comment]
But you cant fhir.truncate(all resources - [StructureDefinition, ValueSet, ConceptMap])

**Comment by mschielmann** [comment]
Yes, but I don't want to know/remember that `ValueSet`, `ConceptMap`, `StructureDefinition` etc. are some special dictionary tables. The whole thing here is about that I don't want to worry with every fhirbase release that something would break the script I've posted above. I don't want to worry if there is a new dictionary table or that some dictionary tables were removed/changed.
As a user I would like to have a function like `truncate(all)` that would clear all changes made by me (user), but at the same time would leave the dictionary tables untouched.
At the moment I don't need any function like `truncate('resourceType')` as I can do `delete from public.resourceType` and `delete from public.resourceType_history` myself, and that is basically what the script does. But if you add another dictionary table, my script would break things (clear this new dictionary table) and so I'll have to look for that new table and change the script.
So that is why I was wondering if this kind of functionality can be incorporated into core fhirbase - so that every time you add new dictionary table a truncate function is updated not to clear it. And from user perspective nothing changes - it is still `fhir.truncate(null)` that clears all resources but dictionary tables.
Or maybe there can be some flag at public.structuredefinition that would indicate that some tables are dictionary and some are not. That would solve it for me as well.
